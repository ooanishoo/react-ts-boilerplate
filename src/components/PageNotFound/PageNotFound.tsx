import React from 'react'

const PageNotFound: React.FC = () => {
    return (
        <section data-testid="page-not-found">
            <h1>Opps! The page you requested was not found.</h1>
        </section>
    )
}

export default PageNotFound
