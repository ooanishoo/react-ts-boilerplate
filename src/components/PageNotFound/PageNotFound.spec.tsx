import { render } from '@testing-library/react'
import React from 'react'
import PageNotFound from './PageNotFound'

describe('<PageNotFound/>', () => {
    test('should render without crashing', () => {
        const expectedOutput = 'Opps! The page you requested was not found.'
        const { container, getByTestId } = render(<PageNotFound />)
        expect(container).toMatchSnapshot()
        expect(getByTestId('page-not-found')).toHaveTextContent(expectedOutput)
    })
})
