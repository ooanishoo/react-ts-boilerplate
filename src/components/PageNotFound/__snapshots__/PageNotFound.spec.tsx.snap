// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageNotFound/> should render without crashing 1`] = `
<div>
  <section
    data-testid="page-not-found"
  >
    <h1>
      Opps! The page you requested was not found.
    </h1>
  </section>
</div>
`;
